music5000.js

```diff
@@ -0,0 +1,195 @@
+"use strict";
+
+// Code ported from Beebem (C to .js) by Jason Robson
+const AUDIO_BUFFER_SIZE = 256;
+
+const RAM_SIZE = 2048;
+const WAVE_TABLE_SIZE = 128;
+const WAVE_TABLES = 14;
+const NUM_CHANNELS = 16;
+const CHANNEL_REG_OFFSET = WAVE_TABLES * WAVE_TABLE_SIZE;
+const CHANNEL_ROW_SIZE = 128;
+
+// Control register bits
+const CTRL_STEREO_POS = (0xf);
+const CTRL_INVERT_WAVE = (1 << 4);
+const CTRL_MODULATE_ADJ = (1 << 5);
+
+// Data bits
+const DATA_SIGN = (1 << 7);
+const DATA_VALUE = (0x7f);
+const FREQ_DISABLE = (1 << 0);
+const NOT_FREQ_DISABLE = 0xfe;
+
+// Channel register sets
+const REG_SET_NORMAL = 0;
+const REG_SET_ALT = 1;
+
+
+export function Music5000(onBuffer) {
+	this._onBufferMusic5000 = onBuffer;
```

Comment:
Can you please use four spaces for indenting like the rest of the codebase? I really should set up automatic formatting -- sorry!

Reply:
That's the issue - the rest of the codebase is indented with 4 spaces, not tabs. Can you use that please? Else I'll need to reformat it. I don't use tabs to indent, I use spaces (which means, rather rudely, I get to pick the indentation size and everyone else has to suffer it...I know that's not idea but I don't fancy mixing tabs and spaces now)